iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -409,10 +409,18 @@ static LogicalResult setRootConfig(
     FuncOp entryPointFn, linalg::ContractionOpInterface contractionOp,
     ArrayRef<LoopTilingAndDistributionInfo> tiledLoops) {
   auto linalgOp = cast<linalg::LinalgOp>(contractionOp.getOperation());
+  // Consider all element types and use the smallest vector size. The tiling
+  // sizes are chosen based on the vector size.
   auto lhsShapedType = contractionOp.lhs().getType().cast<ShapedType>();
+  auto rhsShapedType = contractionOp.rhs().getType().cast<ShapedType>();
```

Comment:
Fantastic! To some extent this kind of logic exists while setting a configuration for the `linalg.generic` op (see in this file below).  It would probably be good to common out this and use it. Also we dont need to use the same `vectorSize` for all dimensions. That might help as well. More of an FYI for you to keep track of. We dont need to do this with this PR.

Reply:
> To some extent this kind of logic exists while setting a configuration for the linalg.generic op (see in this file below). It would probably be good to common out this and use it.

Good point! I'll factor out this and use it in generic cases and matmul cases.

> Also we dont need to use the same vectorSize for all dimensions

Just put a note here. I've considered this. It's tricky. Say that the matmul is mk,kn->mn where input operands are i8 types and result in i32 type. The loop indices are used in both inputs and outputs, which forces us to use the same `vectorSize` for all dimensions. We may have different strategy for quantized cases, because we may load packed inputs in the same time (e.g., load 4 elements from lhs and rhs using 2 registers, and produce 4 i32 values using 4 registers.) I'm looking forward to working on it in the future!